server/channels/app/user.go

```diff
@@ -1775,22 +1776,32 @@ func (a *App) UpdateUserRolesWithUser(c request.CTX, user *model.User, newRoles
 	return ruser, nil
 }
 
-// jobProgressValue defines the types that can be used for job progress values
-type jobProgressValue interface {
+// jobMetadataValue defines the types that can be used for job progress values
+type jobMetadataValue interface {
 	int | string | bool
 }
 
-// updateJobProgress is a generic helper function to update job progress with various value types
-func updateJobProgress[T jobProgressValue](logger mlog.LoggerIFace, store store.Store, job *model.Job, key string, value T) {
+// updateJobProgressWithMetadata is a generic helper function to update job progress with various value types
+func updateJobProgressWithMetadata[T jobMetadataValue](logger mlog.LoggerIFace, jobSrv *jobs.JobServer, job *model.Job, key string, value T) {
 	if job != nil {
 		job.Data[key] = fmt.Sprint(value)
-		if _, err := store.Job().UpdateOptimistically(job, model.JobStatusInProgress); err != nil {
-			logger.Warn("Failed to update job status", mlog.Err(err))
+		if appErr := jobSrv.SetJobProgress(job, 0); appErr != nil {
```

Comment:
This shouldn't set progress always to 0, it should retain the previous progress if any and only set to 0 when it was empty, right?

Reply:
Effectively it's the same thing. Because there was no job progress anyways. It would be 0, and then move to 100 once done. But if I have to handle that, _and_ not duplicate code, that'll be difficult.

See the convo here: https://github.com/mattermost/mattermost/pull/30969#discussion_r2125890739.